pkg/backup/backup.go

```diff
@@ -174,94 +175,188 @@ func (b *Backup) Run() {
 	}
 }
 
+// saveSnap saves the latest snapshot if latest revision is greater than the given lastSnapRev.
 func (b *Backup) saveSnap(lastSnapRev int64) (int64, error) {
-	podList, err := b.kclient.Core().Pods(b.namespace).List(k8sutil.ClusterListOpt(b.clusterName))
+	start := time.Now()
+
+	snap, rev, err := GetSnap(b.kclient, b.etcdTLSConfig, b.namespace, b.clusterName)
 	if err != nil {
-		return lastSnapRev, err
+		return lastSnapRev, fmt.Errorf("failed to get snapshot: (%v)", err)
 	}
+	defer snap.Close()
```

Comment:
call `defer snap.Close()` here because I need to close snap when `if rev <= lastSnapRev {` is true, `etcdVersion, err := snap.Version()` errors out, and `n, err := WriteSnap(etcdVersion, rev, b.be, snap)` errors out.